File: apis/elysium-player-api/src/main/java/com/seventh_root/elysium/api/player/PlayerProvider.java
package com.seventh_root.elysium.api.player;

import com.seventh_root.elysium.core.service.ServiceProvider;

public interface PlayerProvider<T extends ElysiumPlayer> extends ServiceProvider {

    T getPlayer(int id);

    void addPlayer(T player);

    void removePlayer(T player);

}

File: apis/elysium-player-api/src/main/kotlin/com/seventh_root/elysium/api/player/PlayerProvider.kt
package com.seventh_root.elysium.api.player

import com.seventh_root.elysium.core.service.ServiceProvider

interface PlayerProvider<T : ElysiumPlayer> : ServiceProvider {

    fun getPlayer(id: Int): T?
    fun addPlayer(player: T)
    fun removePlayer(player: T)

}
